src/SprykerEco/Yves/Payone/Form/KlarnaSubForm.php

```diff
@@ -0,0 +1,184 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Yves\Payone\Form;
+
+use Generated\Shared\Transfer\PaymentTransfer;
+use Generated\Shared\Transfer\PayoneKlarnaTransfer;
+use Spryker\Yves\StepEngine\Dependency\Form\SubFormInterface;
+use SprykerEco\Shared\Payone\PayoneConstants;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\HiddenType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Validator\Constraints\NotBlank;
+
+class KlarnaSubForm extends AbstractPayoneSubForm
+{
+    protected const PAYMENT_METHOD = 'klarna';
+    protected const FIELD_PAY_METHOD_TYPE = 'payMethod';
+    public const PAY_METHOD_CHOICES = 'pay_methods';
+    public const WIDGET_PAY_METHODS = 'widget_pay_methods';
+    protected const FIELD_PAY_METHOD_TOKENS = 'payMethodTokens';
+    public const BILLING_ADDRESS_DATA = 'billing_address_data';
+    public const CUSTOMER_DATA = 'customer_data';
+    protected const FORM_TEMPLATE_PATH = '%s/%s';
+
+    // billing data
+    public const GIVEN_NAME = 'given_name';
+    public const FAMILY_NAME = 'family_name';
```

Comment:
* Please check the constants inside the form, leave only those that are needed by the form, move the rest to the data provider.

Reply:
++